PowerDNS Admin leads into failure after accepting IDN Subdomain

Hello!

The following Records are working correctly:
whatever.com._report._dmarc    TXT     600    "v=DMARC1;"
xn--domain-mgen-yfb.com.report.dmarc    TXT     600    "v=DMARC1;"

### Problem:
xn--domain-mgen-yfb.com._report._dmarc    TXT     600    "v=DMARC1;"

If there are records like "_report":
- the IDN Domain will be accepted without any error
- also correctly written to pdns
- can also be resolved.
- **But unfortunately PowerDNS Admin is unable to edit the domain, just presents an error 500.**

It looks like **pretty_domain_name** does not correctly split up the Subdomain as soon as there is a Subdomain part starting with "_".

### Debug Log
```
[2022-09-28 14:05:08 +0000] [11] [ERROR] Exception on /domain/testdomain.com [GET]
Traceback (most recent call last):
  File "/app/powerdnsadmin/lib/utils.py", line 252, in pretty_domain_name
    return to_idna(value, 'decode')
  File "/app/powerdnsadmin/lib/utils.py", line 272, in to_idna
    result.append(idna.decode(split))
  File "/usr/lib/python3.8/site-packages/idna/core.py", line 393, in decode
    s = ulabel(label)
  File "/usr/lib/python3.8/site-packages/idna/core.py", line 308, in ulabel
    check_label(label)
  File "/usr/lib/python3.8/site-packages/idna/core.py", line 261, in check_label
    raise InvalidCodepoint('Codepoint {0} at position {1} of {2} not allowed'.format(_unot(cp_value), pos+1, repr(label)))
powerdns_admin_qs | idna.core.InvalidCodepoint: Codepoint U+005F at position 1 of '_report' not allowed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/app/powerdnsadmin/decorators.py", line 75, in decorated_function
    return f(*args, **kwargs)
  File "/app/powerdnsadmin/routes/domain.py", line 130, in domain
    return render_template('domain.html',
  File "/usr/lib/python3.8/site-packages/flask/templating.py", line 137, in render_template
    return _render(
  File "/usr/lib/python3.8/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/usr/lib/python3.8/site-packages/jinja2/environment.py", line 1291, in render
    self.environment.handle_exception()
  File "/usr/lib/python3.8/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/app/powerdnsadmin/templates/domain.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/app/powerdnsadmin/templates/base.html", line 192, in top-level template code
    {% block content %}
  File "/app/powerdnsadmin/templates/domain.html", line 70, in block 'content'
    {{ (record.name,domain.name) | display_record_name | pretty_domain_name }}
  File "/app/powerdnsadmin/lib/utils.py", line 254, in pretty_domain_name
    raise Exception('Cannot decode IDN domain')
Exception: Cannot decode IDN domain
[2022-09-28 14:05:08,812] [app.py:1891] ERROR - Exception on /domain/testdomain.com [GET]
Traceback (most recent call last):
  File "/app/powerdnsadmin/lib/utils.py", line 252, in pretty_domain_name
    return to_idna(value, 'decode')
  File "/app/powerdnsadmin/lib/utils.py", line 272, in to_idna
    result.append(idna.decode(split))
  File "/usr/lib/python3.8/site-packages/idna/core.py", line 393, in decode
    s = ulabel(label)
  File "/usr/lib/python3.8/site-packages/idna/core.py", line 308, in ulabel
    check_label(label)
  File "/usr/lib/python3.8/site-packages/idna/core.py", line 261, in check_label
    raise InvalidCodepoint('Codepoint {0} at position {1} of {2} not allowed'.format(_unot(cp_value), pos+1, repr(label)))
idna.core.InvalidCodepoint: Codepoint U+005F at position 1 of '_report' not allowed

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/lib/python3.8/site-packages/flask_login/utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "/app/powerdnsadmin/decorators.py", line 75, in decorated_function
    return f(*args, **kwargs)
  File "/app/powerdnsadmin/routes/domain.py", line 130, in domain
    return render_template('domain.html',
  File "/usr/lib/python3.8/site-packages/flask/templating.py", line 137, in render_template
    return _render(
  File "/usr/lib/python3.8/site-packages/flask/templating.py", line 120, in _render
    rv = template.render(context)
  File "/usr/lib/python3.8/site-packages/jinja2/environment.py", line 1291, in render
    self.environment.handle_exception()
  File "/usr/lib/python3.8/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/app/powerdnsadmin/templates/domain.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/app/powerdnsadmin/templates/base.html", line 192, in top-level template code
    {% block content %}
  File "/app/powerdnsadmin/templates/domain.html", line 70, in block 'content'
    {{ (record.name,domain.name) | display_record_name | pretty_domain_name }}
  File "/app/powerdnsadmin/lib/utils.py", line 254, in pretty_domain_name
    raise Exception('Cannot decode IDN domain')
Exception: Cannot decode IDN domain
```

--- a/powerdnsadmin/lib/utils.py
+++ b/powerdnsadmin/lib/utils.py
@@ -119,12 +119,16 @@ def fetch_json(remote_url,
 
 
 def display_record_name(data):
-    record_name, domain_name = data
-    if record_name == domain_name:
-        return '@'
+    # Check that the data argument is a tuple containing two elements
+    if isinstance(data, Iterable) and len(data) == 2:
+        record_name, domain_name = data
+        if record_name == domain_name:
+            return '@'
+        else:
+            return record_name
     else:
-        return re.sub('\.{}$'.format(domain_name), '', record_name)
-
+        # If data is not a tuple of length 2, return an empty string
+        return ''
 
 def display_master_name(data):
     """

